src/Ocelot/Requester/HttpClientBuilder.cs

```diff
@@ -1,46 +1,110 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Net;
 using System.Net.Http;
+using System.Threading.Tasks;
 using Ocelot.Logging;
 using Ocelot.Requester.QoS;
 
 namespace Ocelot.Requester
 {
-    internal class HttpClientBuilder
+    internal class HttpClientBuilder : IHttpClientBuilder
     {
-        private readonly Dictionary<int, Func<DelegatingHandler>> _handlers = new Dictionary<int, Func<DelegatingHandler>>();
+        private TimeSpan? _timeout;
+        private readonly List<DelegatingHandler> _handlers = new List<DelegatingHandler>();
+        private Dictionary<string, string> _defaultHeaders;
+        private CookieContainer _cookieContainer;
+        private IQoSProvider _qoSProvider;
 
-        public HttpClientBuilder WithQoS(IQoSProvider qoSProvider, IOcelotLogger logger, HttpMessageHandler innerHandler)
+        public IHttpClientBuilder WithCookieContainer(CookieContainer cookieContainer)
         {
-            _handlers.Add(5000, () => new PollyCircuitBreakingDelegatingHandler(qoSProvider, logger, innerHandler));
+            _cookieContainer = cookieContainer;
             return this;
         }
 
-        internal HttpClient Build(HttpMessageHandler innerHandler)
+        public IHttpClientBuilder WithTimeout(TimeSpan timeout)
```

Comment:
This method isnt currently used, delete it for now and implement when it is needed.